### DISCUSSION TITLE
DBeaver upgrade experience (too many clicks)

### DISCUSSION THREAD

**Discussion by Amitkad** [dbeaver/dbeaver#35088] [General]
Upgrading DBeaver experience today is somewhat cumbersome. Even if everything goes smoothly, it still requires too many clicks and too much effort from the user. My suggestion is to streamline this process into a single-click update.

Here's a story to illustrate my experience with updating DBeaver:

1. I open DBeaver and get prompted to upgrade, so I click yes.
2. I leave to continue my work.
3. After some time, I realize nothing has happened because it only opened the wizard.
4. I click through the "next" buttons until it starts installing.
5. I go back to my work while it installs.
6. After a while, I want to return to DBeaver, but I realize it hasn't restarted because I need to click "finish" on the final screen.
7. I click "finish" and switch back to another tab or screen to do something else while it opens (as it can take a minute some time).
8. After some time, I return and see that nothing has opened.
9. I search for my DBeaver shortcut and open it manually.
10. waiting or working in the meantime.
11. Finally, it opens.

In summary, as a user who multitasks during installations and doesn't sit idle waiting for the process to complete, I want to be able to click "update" and be done with it. Ideally, there could be an option for a custom update with more details, but the main goal is to ensure that when I check back, I have the latest version of DBeaver open and ready to use.

I'd be happy to hear what you think about that?
Especially because DBeaver is getting a lot of upgrades (which is very positive!)
